File: app/src/main/java/com/example/solar_alarm/DisplayModels/SolarTimeDisplayModel.java
package com.example.solar_alarm.DisplayModels;

import android.app.Application;
import android.os.Build;

import androidx.annotation.RequiresApi;

import com.example.solar_alarm.Data.Enums.SolarTimeTypeEnum;
import com.example.solar_alarm.Data.Tables.SolarTime;

import java.time.ZonedDateTime;

@RequiresApi(api = Build.VERSION_CODES.O)
public class SolarTimeDisplayModel {

    private Application _Application;
    private SolarTime _SolarTime;

    public SolarTimeDisplayModel(Application application, SolarTime solarTime)
    {
        _Application = application;
        _SolarTime  = solarTime;
    }


    public ZonedDateTime getSunrise() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.Sunrise);}
    public ZonedDateTime getSunset() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.Sunset);}
    public ZonedDateTime getSolarNoon() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.SolarNoon);}
    public ZonedDateTime getCivilTwilightBegin() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.CivilTwilightBegin);}
    public ZonedDateTime getCivilTwilightEnd() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.CivilTwilightEnd);}
    public ZonedDateTime getNauticalTwilightBegin() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.NauticalTwilightBegin);}
    public ZonedDateTime getNauticalTwilightEnd() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.NauticalTwilightEnd);}
    public ZonedDateTime getAstronomicalTwilightBegin() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.AstronomicalTwilightBegin);}
    public ZonedDateTime getAstronomicalTwilightEnd() throws Exception {return _SolarTime.GetLocalZonedDateTime(SolarTimeTypeEnum.AstronomicalTwilightEnd);}
}

File: app/src/main/java/com/example/solar_alarm/DisplayModels/SolarTimeDisplayModel.kt
package com.example.solar_alarm.displayModels

// Repair: Fixed broken package/import lines
import androidx.annotation.RequiresApi
import android.os.Build
import android.app.Application
import com.example.solar_alarm.data.tables.SolarTime
import kotlin.Throws
import com.example.solar_alarm.data.enums.SolarTimeTypeEnum
import java.lang.Exception
import java.time.ZonedDateTime

@RequiresApi(api = Build.VERSION_CODES.O)
class SolarTimeDisplayModel(private val _Application: Application, private val _SolarTime: SolarTime?) {
    @get:Throws(Exception::class)
    val sunrise: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.Sunrise)

    @get:Throws(Exception::class)
    val sunset: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.Sunset)

    @get:Throws(Exception::class)
    val solarNoon: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.SolarNoon)

    @get:Throws(Exception::class)
    val civilTwilightBegin: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.CivilTwilightBegin)

    @get:Throws(Exception::class)
    val civilTwilightEnd: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.CivilTwilightEnd)

    @get:Throws(Exception::class)
    val nauticalTwilightBegin: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.NauticalTwilightBegin)

    @get:Throws(Exception::class)
    val nauticalTwilightEnd: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.NauticalTwilightEnd)

    @get:Throws(Exception::class)
    val astronomicalTwilightBegin: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.AstronomicalTwilightBegin)

    @get:Throws(Exception::class)
    val astronomicalTwilightEnd: ZonedDateTime
        get() = _SolarTime!!.getLocalZonedDateTime(SolarTimeTypeEnum.AstronomicalTwilightEnd)
}